packages/wasm/src/build.ts

```diff
@@ -24,17 +30,32 @@ export const witness = (txPlan: TransactionPlan, sct: StateCommitmentTree): Witn
   return WitnessData.fromJsonString(JSON.stringify(result));
 };
 
-export const build = async (
-  fullViewingKey: string,
+export const buildParallel = (
+  batchActions: Action[],
   txPlan: TransactionPlan,
   witnessData: WitnessData,
   authData: AuthorizationData,
-): Promise<Transaction> => {
-  await loadProvingKeys();
-
+): Transaction => {
   const result = validateSchema(
     WasmBuildSchema,
-    wasmBuild(fullViewingKey, txPlan.toJson(), witnessData.toJson(), authData.toJson()),
+    build_parallel(batchActions, txPlan.toJson(), witnessData.toJson(), authData.toJson()),
   );
-  return Transaction.fromJsonString(JSON.stringify(result));
+
+  return Transaction.fromJson(result as JsonValue);
+};
+
+export const buildActionParallel = (
+  txPlan: TransactionPlan,
+  witnessData: WitnessData,
+  fullViewingKey: string,
+  actionId: number,
+): Action => {
+  const result = build_action(
+    txPlan.toJson(),
+    txPlan.actions[actionId]?.toJson(),
+    fullViewingKey,
+    witnessData.toJson(),
+  ) as Action;
```

Comment:
this cast seems incorrect. we are calling the wasm build_action directly and i don't believe it's returning a real Action, but some json that hopefully represents an action.

Reply:
https://github.com/penumbra-zone/penumbra/blob/main/crates/wasm/src/build.rs#L43 right?

Reply:
confirmed the output from calling `buildAction` is returning the JsonValue of `Action`.

Reply:
This is tricky. Some actions have the `authSig` field (spend action), while others do not (output action). `authSig` is currently set as a placeholder value I believe

<img width="606" alt="Screenshot 2023-12-22 at 1 53 40 PM" src="https://github.com/penumbra-zone/web/assets/70081547/aa305c8d-8dc6-4856-87f1-2c45c2e32399">

Reply:
ok, looked this up for my own clarity: the transaction as returned is unauthorized, and so `authSig` is *expected* to be absent at this stage of the plan/build/auth flow, and this is not a failure.

https://github.com/penumbra-zone/penumbra/blob/1eb74b7c78e114759a270a358e36a7b11cf117eb/crates/wasm/src/build.rs#L41C39-L41C39

but i would like to issue the wasm crate to not define `authSig`, rather than define junk.

Reply:
> ok, looked this up for my own clarity: the transaction as returned is unauthorized, and so `authSig` is _expected_ to be absent at this stage of the plan/build/auth flow, and this is not a failure.

If you look at Henry's comment in the [split-build](https://github.com/penumbra-zone/penumbra/pull/3289) PR, you'll notice that defining dummy `authSig` was an intentional design choice, although it may be unappealing.

Reply:
Resolved using `zod record`